File: JANUARY/14-01-2026/Police and Thieves.py
class Solution:
    def catchThieves(self, arr, k):
        police = []
        thieves = []
        n = len(arr)
        caught = 0

        for i in range(n):
            if arr[i] == 'P':
                police.append(i)
            elif arr[i] == 'T':
                thieves.append(i)

        i = j = 0
        while i < len(police) and j < len(thieves):
            if abs(police[i] - thieves[j]) <= k:
                caught += 1
                i += 1
                j += 1
            elif police[i] < thieves[j]:
                i += 1
            else:
                j += 1

        return caught

File: JANUARY/14-01-2026/Police and Thieves.java
class Solution {
    public int catchThieves(char[] arr, int k) {
        List<Integer> police = new ArrayList<>();
        List<Integer> thieves = new ArrayList<>();
        int n = arr.length, caught = 0;

        for (int i = 0; i < n; i++) {
            if (arr[i] == 'P') police.add(i);
            else if (arr[i] == 'T') thieves.add(i);
        }

        int i = 0, j = 0;
        while (i < police.size() && j < thieves.size()) {
            if (Math.abs(police.get(i) - thieves.get(j)) <= k) {
                caught++;
                i++;
                j++;
            } else if (police.get(i) < thieves.get(j)) {
                i++;
            } else {
                j++;
            }
        }

        return caught;
    }
}